scripts/release/debian/deb_install.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+#######################################################################################################################
+# This script does three fundamental things:                                                                          #
+#   1. Add Microsoft's GPG Key has a trusted source of apt packages.                                                  #
+#   2. Add Microsoft's repositories as a source for apt packages.                                                     #
+#   3. Installs the Azure CLI from those repositories.                                                                #
+# Given the nature of this script, it must be executed with elevated privileges, i.e. with `sudo`.                    #
+#                                                                                                                     #
+# Remember, with great power comes great responsibility.                                                              #
+#                                                                                                                     #
+# Do not be in the habit of executing scripts from the internet with root-level access to your machine. Only trust    #
+# well-known publishers.                                                                                              #
+#######################################################################################################################
+
+set -e
+
+if [[ $# -ge 1 && $1 == "-y" ]]; then
+    global_consent=0
+else
+    global_consent=1
+fi
+
+function assert_consent {
+    if [[ $2 -eq 0 ]]; then
+        return 0
+    fi
+
+    echo -n "$1 [Y/n] "
+    read consent
+    if [[ ! "${consent}" == "y" && ! "${consent}" == "Y" && ! "${consent}" == "" ]]; then
+        echo "'${consent}'"
+        exit 1
+    fi
+}
+
+global_consent=0 # Artificially giving global consent after review-feedback. Remove this line to enable interactive mode
+
+assert_consent "Add packages necessary to modify your apt-package sources?" ${global_consent}
+set -v
+apt-get update
+apt-get install -y apt-transport-https lsb-release gnupg curl
```

Comment:
Do need `curl` since the whole flow starts with a `curl`? I am fine to let users pre-install curl.

Reply:
I ended up including it because I figured there's a chance that people download the script using their browser to inspect it, then run the script. Because I use curl a few lines down, I figured its safest to assert its presence.